packages/extension-api/src/extension-api.d.ts

```diff
@@ -306,4 +313,39 @@ declare module '@tmpwip/extension-api' {
      */
     affectsConfiguration(section: string, scope?: ConfigurationScope): boolean;
   }
+
+  export type DialogType = 'none' | 'info' | 'error' | 'question' | 'warning';
+
+  /**
+   * Defines a generalized way of reporting progress updates.
+   */
+  export interface Progress<T> {
+    /**
+     * Report a progress update.
+     * @param value A progress item, like a message and/or an
+     * report on how much work finished
+     */
+    report(value: T): void;
+  }
+
+  export namespace window {
+    /**
+     * Show an dialog window. Optionally provide an array of items which will be presented as
+     * clickable buttons.
+     *
+     * @param message The message to show.
+     * @param items A set of items that will be rendered as actions in the message.
+     * @return A thenable that resolves to the selected item or `undefined` when being dismissed.
+     */
+    export function showDialog(
```

Comment:
I would keep vs code like pattern

showInformationMessage / showWarningMessage / showErrorMessage

Reply:
and use the same args